File: frontend/src/types/sale.ts
// backend/app/schemas/sale.py ile birebir eşleşir.
export interface SaleItemIn {
  product_id: number;
  quantity: number;
}

export interface SaleItemOut {
  product_id: number;
  quantity: number;
  unit_price: number;
}

export interface SaleOut {
  id: number;
  branch_id: number;
  items: SaleItemOut[];
  total: number;
  payment_method: string;
  status: string;
  created_at: string;
}

export interface SaleListItem {
  id: number;
  sale_date: string;
  total: number;
  payment_method: string;
}

export interface SaleDetailItem {
  product_id: number;
  product_name: string;
  quantity: number;
  unit_price: number;
  returnable_quantity: number;
}

export interface SaleDetail {
  id: number;
  sale_date: string;
  branch_id: number;
  total: number;
  payment_method: string;
  items: SaleDetailItem[];
}

File: backend/app/schemas/sale.py
from datetime import datetime

from pydantic import BaseModel


class SaleItemIn(BaseModel):
    product_id: int
    quantity: int


class SaleCreate(BaseModel):
    items: list[SaleItemIn]
    payment_method: str


class SaleItemOut(BaseModel):
    product_id: int
    quantity: int
    unit_price: float


class SaleOut(BaseModel):
    id: int
    branch_id: int
    items: list[SaleItemOut]
    total: float
    payment_method: str
    status: str
    created_at: datetime


class SaleListItem(BaseModel):
    id: int
    sale_date: datetime
    total: float
    payment_method: str


class SaleDetailItem(BaseModel):
    product_id: int
    product_name: str
    quantity: int
    unit_price: float
    returnable_quantity: int


class SaleDetail(BaseModel):
    id: int
    sale_date: datetime
    branch_id: int
    total: float
    payment_method: str
    items: list[SaleDetailItem]
